src/Microsoft.TemplateEngine.Cli/HelpAndUsage/TemplateDetailsDisplay.cs

```diff
@@ -220,30 +219,35 @@ private static void ShowParameterHelp(IReadOnlyDictionary<string, string> inputP
                     // display the default value if there is one
                     if (!string.IsNullOrEmpty(param.DefaultValue))
                     {
-                        if (param is IAllowDefaultIfOptionWithoutValue paramWithNoValueDefault
-                            && !string.IsNullOrEmpty(paramWithNoValueDefault.DefaultIfOptionWithoutValue)
-                            && !string.Equals(paramWithNoValueDefault.DefaultIfOptionWithoutValue, param.DefaultValue, StringComparison.Ordinal))
+                        displayValue.AppendLine(string.Format(LocalizableStrings.DefaultValue, param.DefaultValue));
+                    }
+
+                    if (param is IAllowDefaultIfOptionWithoutValue paramWithNoValueDefault
+                        && !string.IsNullOrEmpty(paramWithNoValueDefault.DefaultIfOptionWithoutValue))
+                    {
+                        // default if option is provided without a value should not be displayed if:
+                        // - it is bool parameter with "DefaultIfOptionWithoutValue": "true"
+                        // - it is not bool parameter (int, string, etc) and default value coincides with "DefaultIfOptionWithoutValue"
+                        if (string.Equals(param.DataType, "bool", StringComparison.OrdinalIgnoreCase))
                         {
-                            // the regular default, and the default if the switch is provided without a value
-                            displayValue.AppendLine(string.Format(LocalizableStrings.DefaultValuePlusSwitchWithoutValueDefault, param.DefaultValue, paramWithNoValueDefault.DefaultIfOptionWithoutValue));
-                            anyParamsShowingDefaultForSwitchWithNoValue = true;
+                            if (!string.Equals(paramWithNoValueDefault.DefaultIfOptionWithoutValue, "true", StringComparison.OrdinalIgnoreCase))
+                            {
+                                displayValue.AppendLine(string.Format(LocalizableStrings.DefaultIfOptionWithoutValue, paramWithNoValueDefault.DefaultIfOptionWithoutValue));
```

Comment:
please consider string concatenation instead of string.Format, if it is equivalently easy to maintain.  For reference, look at the IL for both concatenation and Format.

Reply:
I do not want to switch formatting to concatenation because this will give less freedom for the guys who localize strings. Now they have a parameter of a localizable string and they could put it everywhere they wish in the string. Maybe, in this particular situation, it is almost safe, and in every language, we could put the parameter at the end of a string, but I do not want to start such a practice.